src/main/java/no/nav/finnkandidatapi/kafka/harTilretteleggingsbehov/handler/KandidatHandler.java

```diff
@@ -0,0 +1,57 @@
+package no.nav.finnkandidatapi.kafka.harTilretteleggingsbehov.handler;
+
+import com.nimbusds.oauth2.sdk.util.CollectionUtils;
+import lombok.extern.slf4j.Slf4j;
+import no.nav.finnkandidatapi.kafka.harTilretteleggingsbehov.HarTilretteleggingsbehov;
+import no.nav.finnkandidatapi.kafka.harTilretteleggingsbehov.HarTilretteleggingsbehovProducer;
+import no.nav.finnkandidatapi.kafka.harTilretteleggingsbehov.SammenstillBehov;
+import no.nav.finnkandidatapi.kandidat.Kandidat;
+import no.nav.finnkandidatapi.metrikker.KandidatEndret;
+import no.nav.finnkandidatapi.metrikker.KandidatOpprettet;
+import no.nav.finnkandidatapi.metrikker.KandidatSlettet;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+
+@Component
+@Slf4j
+public class KandidatHandler {
+    private SammenstillBehov sammenstillBehov;
+    private HarTilretteleggingsbehovProducer harTilretteleggingsbehovProducer;
+
+    public KandidatHandler(
+            SammenstillBehov sammenstillBehov, HarTilretteleggingsbehovProducer harTilretteleggingsbehovProducer) {
+        this.sammenstillBehov = sammenstillBehov;
+        this.harTilretteleggingsbehovProducer = harTilretteleggingsbehovProducer;
+    }
+
+    @EventListener
+    public void kandidatOpprettet(KandidatOpprettet event) {
+        Kandidat kandidat = event.getKandidat();
+        kandidatOpprettetEllerEndret(kandidat);
+    }
+
+    @EventListener
+    public void kandidatEndret(KandidatEndret event) {
+        Kandidat kandidat = event.getKandidat();
+        kandidatOpprettetEllerEndret(kandidat);
+    }
+
+    @EventListener
+    public void kandidatSlettet(KandidatSlettet event) {
+        harTilretteleggingsbehovProducer.sendKafkamelding(
+                sammenstillBehov.lagbehovKandidat(
+                        new HarTilretteleggingsbehov(event.getAktørId(), false, Collections.emptyList()))
+        );
+    }
+
+    private void kandidatOpprettetEllerEndret(Kandidat kandidat) {
+        HarTilretteleggingsbehov behov = sammenstillBehov.lagbehovKandidat(
+                new HarTilretteleggingsbehov(
+                        kandidat.getAktørId(),
+                        CollectionUtils.isNotEmpty(kandidat.kategorier()),
```

Comment:
Er det enklere å bare bruke `!kandidat.kategorier().isEmpty()` her?

Reply:
Hmm, hadde jeg bare hatt tilgjengelig en hasValue metode...

Om jeg slår på stemmen inne i hodet mitt så synes jeg at "kandidat kategori ikke tom" leses/tolkes bedre inne i hodet, enn "ikke kandidat kategori er tom". 

Det føles som dobbel negasjon uansett da(ville heller hatt hasValue eller noe slikt), men i det siste tilfellet blir den samlet, i steden for spredt... 

! blir av og til oversett har jeg merket meg, vi har hatt feil i koden tidligere pga at tegnet har blitt oversett i forbindelse med at uttrykket har en isEmpty.

Reply:
Men dette er helt klart bare en preferanse jeg har pga at jeg synes noe mangler i språket for å lage konsist uttrykk, og kan endre om dere vil det.

Reply:
Godt poeng at det er mer forklarende og lesbart slik det er nå. Bare behold det slik :)